lib/balance-tx/lib/Cardano/Wallet/Shelley/Compatibility/Ledger.hs

```diff
@@ -295,7 +296,10 @@ instance Convert Address (Ledger.Addr StandardCrypto) where
             [ "toLedger @Address: Invalid address:"
             , pretty (Address bytes)
             ]
-    toWallet = Address . Ledger.serialiseAddr
+    toWallet = toWalletAddress
+
+toWalletAddress :: Ledger.Addr StandardCrypto -> Address
```

Comment:
What's wrong with using the existing `toWallet`?

Reply:
> What's wrong with using the existing `toWallet`?

There's nothing wrong with `toWallet`, but in some situations using it requires an additional type application to disambiguate types. We could avoid this type application by changing the functional dependency to be bidirectional.